### ISSUE TITLE
I need demo access

### ISSUE THREAD

**Issue by gsdavris** [VivaPayments/API#2182]
Hello i want to set up a smart checkout in a react (nextjs) app.
I ve made a demo account and I m trying to request access token in 
https://demo-accounts.vivapayments.com/connect/token endpoint as the documentation says but it doesn't work (??)

**Comment by can3ry** [comment]
Hello @gsdavris 

Thank you for your query.

Could you please share the request headers, payload, and response? 

Thanks
Viva Wallet Development Team

**Comment by gsdavris** [comment]
1. Request URL:
      https://demo-accounts.vivapayments.com/connect/token
      2. Referrer Policy:
      strict-origin-when-cross-origin
      1. Request Headers
      1. Provisional headers are shown
      Learn more
      2. Accept:
      application/json, text/plain, */*
      3. Authorization:
      Basic
      ZTQ3NjFrYTdtc3gza2tuamplYWV3cjBrdXB6cmRwYXRwMmxnZDQ3NzZtazcwLmFwcHMudml2YXBheW1lbnRzLmNvbTpidmJRTWs3c1FpdWtuR001YzdMcksycTY3ODlMU0Y=
      4. Content-Type:
      application/x-www-form-urlencoded
      5. Referer:
      https://od5xiz.sse.codesandbox.io/
      6. sec-ch-ua:
      " Not A;Brand";v="99", "Chromium";v="101", "Microsoft Edge";v="101"
      7. sec-ch-ua-mobile:
      ?0
      8. sec-ch-ua-platform:
      "Windows"
      9. User-Agent:
      Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML,
      like Gecko) Chrome/101.0.4951.64 Safari/537.36 Edg/101.0.1210.53



*Payload:*


   1. grant_type=client_credentials:


*Response:*

         *Nothing*




Στις Πέμ 26 Μαΐ 2022 στις 8:57 μ.μ., ο/η can3ry ***@***.***>
έγραψε:

**Comment by can3ry** [comment]
Hello @gsdavris 

I've checked the link https://od5xiz.sse.codesandbox.io/ and noticed that the payload is set as grant_type%3Dclient_credentials in the request, and this causes an error.  If you are sending a raw body, is it possible to send a URL-decoded body?

The below cURL request is exported from the browser(od5xiz.sse.codesandbox.io). 
If I change the data from grant_type%3Dclient_credentials to grant_type=client_credentials, it works.

I hope this helps and please let us know if you need further support.

curl 'https://demo-accounts.vivapayments.com/connect/token' \
  -H 'sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="101", "Google Chrome";v="101"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'Authorization: Basic ZTQ3NjFrYTdtc3gza2tuamplYWV3cjBrdXB6cmRwYXRwMmxnZDQ3NzZtazcwLmFwcHMudml2YXBheW1lbnRzLmNvbTpidmJRTWs3c1FpdWtuR001YzdMcksycTY3ODlMU0Y=' \
  -H 'Content-Type: application/x-www-form-urlencoded' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Referer: https://od5xiz.sse.codesandbox.io/' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.64 Safari/537.36' \
  -H 'sec-ch-ua-platform: "macOS"' \
  --data-raw 'grant_type%3Dclient_credentials' \
  --compressed
`
![PAYLOAD](https://user-images.githubusercontent.com/51880555/170681609-1d29bf9b-d4d9-4cae-9b84-29fdfce17be2.png)

Best Regards
Viva Wallet Development Team

**Comment by gsdavris** [comment]
Thanks for your reply

I ve change the %3D to = and it still doesn't work,

But the strange thing is that it doesn't return a response at all

Στις Παρ, 27 Μαΐ 2022, 13:31 ο χρήστης can3ry ***@***.***>
έγραψε:

**Comment by can3ry** [comment]
Hi @gsdavris 

A further check showed me the below error. 

Access to XMLHttpRequest at 'https://demo-accounts.vivapayments.com/connect/token' from origin 'https://od5xiz.sse.codesandbox.io' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

You may find more information about the CORS policy on the below pages. 

https://blog.container-solutions.com/a-guide-to-solving-those-mystifying-cors-issues
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

This is just another warning message: I see that the API call is being made from the front-end, so the headers and credentials are available to the public. I assume that you will take precautions for this. 

I am closing this ticket for now. Please feel free to open a new ticket if you have further questions.

Best Regards
Viva Wallet Development Team

**Comment by gsdavris** [comment]
Thank you very much for your time,

So as my title says .... Viva have to provide me access to their Api,....
but i already have created a demo account and i already pass my url....

Ps The fontend calls is for development-debugging purposes only.



Στις Παρ, 27 Μαΐ 2022, 15:01 ο χρήστης can3ry ***@***.***>
έγραψε:

**Comment by can3ry** [comment]
@gsdavris

Thank you.

You already have access to the Viva Wallet API. Your credentials work fine, and you can create smart checkout without any issues.
 
Please use the backend to get a smart checkout order code and pass it to the front-end or use a similar approach.

We cannot add origins to our APIs, so please avoid requests that use the CORS.

Please let us know if you need further information.

Best Regards
Viva Wallet Development Team
